larq/metrics.py

```diff
@@ -72,35 +72,32 @@ class FlipRatio(tf.keras.metrics.Metric):
 
     !!! example
         ```python
-        m = metrics.FlipRatio(values_shape=(2,))
+        m = metrics.FlipRatio()
         m.update_state((1, 1))  # result: 0
         m.update_state((2, 2))  # result: 1
         m.update_state((1, 2))  # result: 0.75
         print('Final result: ', m.result().numpy())  # Final result: 0.75
         ```
 
     # Arguments
+    name: Name of the metric.
     values_shape: Shape of the tensor for which to track changes.
     values_dtype: Data type of the tensor for which to track changes.
-    name: Name of the metric.
     dtype: Data type of the moving mean.
     """
 
     def __init__(
-        self, values_shape=(), values_dtype="int8", name="flip_ratio", dtype=None
+        self, values_shape=None, values_dtype="int8", name="flip_ratio", dtype=None
     ):
         super().__init__(name=name, dtype=dtype)
+
         self.values_dtype = tf.as_dtype(values_dtype)
-        self.values_shape = tf.TensorShape(values_shape).as_list()
+        self.values_shape = values_shape
+
         self.is_weight_metric = True
+        self.built = False
```

Comment:
I would prefer to do the same as the `MeanTensor` metric and use `self._built = False` here since `self.build` is an attribute on the keras layer that might change behaviour in an non obvious way.

Reply:
I have it as a public property now because quantizers need to access it to throw away flip ratios on activations, but I could have them read a private `_built` if you prefer instead.